book/src/dev/rfcs/0004-asynchronous-script-verification.md

```diff
@@ -82,18 +91,157 @@ parallelism.
 # Reference-level explanation
 [reference-level-explanation]: #reference-level-explanation
 
-We add a `Request::AwaitUtxo(OutPoint)` and
-`Response::Utxo(transparent::Output)` to the state protocol. As described
-above, the request name is intended to indicate the request's behavior: the
-request does not resolve until the state layer learns of a UTXO described by
-the request.
+## Data structures
+[data-structures]: #data-structures
+
+We add the following request and response to the state protocol:
+```rust
+enum Request::AwaitSpendableUtxo {
+    outpoint: OutPoint,
+    spend_height: Height,
+    spend_pools: SpendPools,
+}
+
+enum SpendPools {
+    SomeTransparentOutputs,
+    AllShieldedOutputs,
+}
+
+enum Response::Utxo(transparent::Output)
+```
+
+As described above, the request name is intended to indicate the request's behavior.
+The request does not resolve until:
+- the state layer learns of a UTXO described by the request, and
+- the output is spendable at `height`, to `spend_output_pools`.
+
+We also add a coinbase flag and height to `transparent::Output`s that we look up in
+the state, or get from newly commited blocks:
+```rust
+enum TransparentOutputOrigin {
+    Coinbase {
+        output: transparent::Output,
+        coinbase_height: Height,
+    },
+    NonCoinbase {
+        output: transparent::Output,
+    },
+}
+```
+
+The coinbase flag can be derived from the `OutPoint.index`.
+There is exactly one coinbase transaction per block, and it must have an index of `0`.
+
+The coinbase height can be:
+- looked up from the `OutPoint.hash` in `tx_by_hash`, or
+- retrieved from the newly arrived block.
+
+## Transparent coinbase consensus rules
+[transparent-coinbase-consensus-rules]: #transparent-coinbase-consensus-rules
+
+Specifically, if the UTXO is a transparent coinbase output,
+the service is not required to return a response if:
+- `height` is less than `MIN_TRANSPARENT_COINBASE_MATURITY` after the `OutPoint`'s height, or
```

Comment:
Suggestion: Maybe we should mention that `MIN_TRANSPARENT_COINBASE_MATURITY == 100` somewhere?

Reply:
Fixed in d014ccd2